backend/infrahub/services/adapters/workflow/worker.py

```diff
@@ -46,7 +48,18 @@ async def execute_workflow(
         expected_return: type[Return] | None = None,
         parameters: dict[str, Any] | None = None,
     ) -> Any:
-        response: FlowRun = await run_deployment(name=workflow.full_name, poll_interval=1, parameters=parameters or {})  # type: ignore[return-value, misc]
+        tags = []
+        if (
+            workflow.branch_support == BranchSupportType.AWARE
+            and isinstance(parameters, dict)
+            and ("branch" in parameters or "branch_name" in parameters)
+        ):
+            branch_name = parameters.get("branch", None) or parameters.get("branch_name")
```

Comment:
After more consideration, I've removed this code and I created a function that can be executed within the flow itself to dynamically add the tag based on the name of the branch. I think it's cleaner and it guarantees that the tag will always be present

I also added a more generic parameter `tags` to `submit_workflow` and `execute_workflow`, I'm sure it will be useful at some point